noxfile.py

```diff
@@ -76,7 +76,7 @@ def activate_virtualenv_in_precommit_hooks(session: Session) -> None:
         hook.write_text("\n".join(lines))
 
 
-@nox.session(name="pre-commit", python="3.9")
+@nox.session(name="pre-commit", python=python_versions)
```

Comment:
Is there a reason why we need to do `pre-commit` with all python versions? I think 3.9 should be fine.

Reply:
Certain libraries like TF1 is only supported by older python versions, so in the long run it would be beneficial to ensure at least some backward compatiability.

Reply:
nice ok makes sense =]

Reply:
Oh you are talking about pre-commit, I see. Nvm. I think we only need tests to run for all python? I have updated this to 3.9 so that CI could run faster.